tools/check-enforcer/Azure.Sdk.Tools.CheckEnforcer/RepositoryConfiguration.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using YamlDotNet.Serialization;
+
+namespace Azure.Sdk.Tools.CheckEnforcer
+{
+    public class RepositoryConfiguration : IRepositoryConfiguration
+    {
+        public RepositoryConfiguration()
+        {
+            MinimumCheckRuns = 1;
+        }
+
+        [YamlMember(Alias = "minimumCheckRuns")]
+        public uint MinimumCheckRuns { get; internal set; }
+
+        [YamlMember(Alias = "enabled")]
+        public bool IsEnabled { get; internal set; }
+
+        [YamlMember(Alias = "format")]
```

Comment:
Perhaps I've missed it but I don't see format used anywhere yet, nor do I see any way to specify jobs to ignore.

Reply:
That's coming. For now for all our internal repos, setting a minimum check of 2 would actually serve the purpose. But I am adding the rules based mechanism. The format is used in the  ```ConfigurationStore``` class.

The intention here is that I may introduce incompatible changes into the format in the future. So I want to get ahead of that and force people to specify what format of configuration file they are using so that I can process it specially.

The ```IRepositoryConfiguration``` interface defines the internal contract for the function for configuration values. At the moment I've got a single ```RepositoryConfiguration``` class that represents the ```v0.1-alpha``` configuration format. I don't think that the format will change in breaking ways much, but its pretty cheap insurance against that becoming a nightmare to manage (allows staged rollout etc).

Reply:
Oh - one of the reasons I haven't implemented the rules yet is that there is that ongoing incident where there is not a check being created which corresponds to the pipeline name - depending on whether that is a bug or the new normal I'll need to change my approach (hopefully it is a bug).

Reply:
I'd like to get this merged now regardless because this can already provide some value in its current state.

Reply:
>That's coming. For now for all our internal repos, setting a minimum check of 2 would actually serve the purpose. But I am adding the rules based mechanism. The format is used in the ConfigurationStore class.

Yeah I guess that make since for now.